### PR TITLE
python3Packages.hdbscan: Patch out runtime Cython dependency

### PR DESCRIPTION
**Pull request by gmacon**
###### Description of changes

Upstream declares a dependency on Cython at runtime, but that's not a true requirements. Remove it so that derivations depending on this one can build without pip complaining that Cython cannot be found.

Building an internal `buildPythonPackage` derivation that depends on hdbscan results in this error:

```
ERROR: Could not find a version that satisfies the requirement cython>=0.27 (from hdbscan) (from versions: none)
ERROR: No matching distribution found for cython>=0.27
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

hdbscan's `checkPhase` indicates that everything is OK. I've run the tests of my internal package with this change overlayed in, and those tests all passed, too.

### CODE REVIEW

File: pkgs/development/python-modules/hdbscan/default.nix [hunk 1]
```diff
@@ -27,6 +27,9 @@ buildPythonPackage rec {
       sha256 = "07d7jdwk0b8kgaqkifd529sarji01j1jiih7cfccc5kxmlb5py9h";
     })
   ];
+  # hdbscan declares a runtime dependency on Cython, but it isn't really necessary.
+  # Patch it out to allow packages depending on hdbscan to build.
+  prePatch = "sed -i~ '/cython/d' requirements.txt";
```
Comment (FRidh) [inline pkgs/development/python-modules/hdbscan/default.nix:32]: please use `postPatch`. That way patches can always apply to the original source.

Comment (gmacon) [reply pkgs/development/python-modules/hdbscan/default.nix]: OK, done.

**Comment by FRidh** [comment]
Come to think of it, we actually have a hook for this as well:
```nix
  pythonRelaxDeps = [
    "cython"
  ];

  nativeBuildInputs = [
    pythonRelaxDepsHook
  ];
```
No need to change from my point of view.

**Comment by gmacon** [comment]
While converting my internal project to use the hook, I discovered some more flaky tests. I reported them upstream and added a commit here disabling them.

**Pull request merged**